spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/filter/factory/cache/keygenerator/KeyValueGenerator.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2013-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gateway.filter.factory.cache.keygenerator;
+
+import java.util.function.Function;
+
+import org.springframework.http.server.reactive.ServerHttpRequest;
+
+/**
+ * @author Marta Medio
```

Comment:
Add a description sentence. Should also not that if there is not value to return empty string?

Reply:
About the empty String, that's something we are currently doing in all the implementations. Do you think we should add a default implementation, or maybe an abstract class to prevent null?

Reply:
yes, or handle null

Reply:
See f779d16e8ee646b74a1dd085ff5c9cae8f4255dc

Reply:
Nice, thank you so much!